src/components/menu/menu-full-screen/menu-full-screen.component.js

```diff
@@ -25,6 +33,9 @@ const MenuFullscreen = ({
   const menuWrapperRef = useRef();
   const menuContentRef = useRef();
   const { menuType } = useContext(MenuContext);
+  const [isAnimationComplete, setAnimationComplete] = useState(false);
```

Comment:
Do we actually need this? Is there any issue with the MenuFullScreen being already focused during the 300ms that the transition takes? Looks like a lot of unnecessary code that does not solve any real problem here

Reply:
I had a lot of problems with getting this to work on MenuFullScreen as that's implemented quite differently from the other components here. The problem is that it uses FocusTrap but not Modal, but FocusTrap sort of assumes it's inside a Modal and in particular uses ModalContext, including isAnimationComplete from that to decide whether to focus on opening or not, which is why I ended up with this sort of messy solution of just putting the ModalContext.Provider into MenuFullScreen (as I understand why it can't use Modal directly). Really what this code is about is ensuring the menu gets the automatic focus at the write time - previously this was done in a rather simple way by putting tabindex of -1 on the container of the actual menu (not including the header) and directly focusing that on open, but now we want to focus the container first and reuse the logic that's currently (in this PR) in useModalFocus it becomes much more tricky.

Ideally these two components (Modal and FocusTrap) should be properly decoupled but it's not clear if that's even possible.

I'm currently looking at all this again anyway to address the issue that Ed raised in his comments so it will likely change a lot - certainly the code is getting way more complex than I would like so I'll bear your comments in mind and see if anything can be removed! But I'm not particularly optimistic.